future/src/test/java/com/spotify/hamcrest/future/ExceptionallyCompletedBlockingCompletionStageTest.java

```diff
@@ -85,4 +88,34 @@ public void testInterruptedMismatchFormatting() throws Exception {
       Thread.interrupted();
     }
   }
+
+  @Test
+  public void testInterruptActuallyInterruptsAssert() throws Exception {
+    final CountDownLatch beforeAssertLatch = new CountDownLatch(1);
+    final CountDownLatch afterAssertLatch = new CountDownLatch(1);
+    final Thread threadToInterrupt = new Thread(() -> {
+      try {
+        beforeAssertLatch.countDown();
+        assertThat(new CompletableFuture<String>(),
+            SUT);
+      } finally {
+        afterAssertLatch.countDown();
+      }
+    });
+    try {
+      threadToInterrupt.setDaemon(true);
+      threadToInterrupt.start();
+
+      beforeAssertLatch.await();
+      TimeUnit.SECONDS.sleep(1);
```

Comment:
This sleep feels like it's a bit unnecessary. Is the point to get the thread 'inside' the assert? It probably deserves an explanation.